addons/stock/static/src/js/report_stock_reception.js

```diff
@@ -0,0 +1,214 @@
+            quantities.push(parseFloat(node.getAttribute('qty')));
+            inIds.push(JSON.parse(node.getAttribute('move-ins-ids')));
+            this._switchButton(node);
+        });
+
+        return this._rpc({
+            model: 'report.stock.report_reception',
+            args: [false, moveIds, quantities, inIds],
+            method: 'action_assign'
+        })
+    },
+
+    /**
+     * Unassign the specified move
+     *
+     * @returns {Promise}
+     */
+     _onClickUnassign: function(ev) {
+        this._switchButton(ev.currentTarget);
+        let quantity = parseFloat(ev.target.getAttribute('qty'));
+        let modelId = parseInt(ev.target.getAttribute('move-id'));
+        let inIds = JSON.parse("[" + ev.target.getAttribute('move-ins-ids') + "]");
+        ev.target.closest('td').nextElementSibling.querySelectorAll('.o_print_label').forEach(button => button.setAttribute('disabled', true));
+        return this._rpc({
+            model: 'report.stock.report_reception',
+            args: [false, modelId, quantity, inIds[0]],
+            method: 'action_unassign'
+        })
+    },
+
+    /**
+     * Open the forecast report for the product of
+     * the selected move
+     */
+    _onClickForecastReport: function(ev) {
+        const model = ev.target.getAttribute('model');
+        const modelId = parseInt(ev.target.getAttribute('move-id'));
+        return this._rpc( {
+            model,
+            args: [[modelId]],
+            method: 'action_product_forecast_report'
+        }).then((action) => {
+            return this.do_action(action);
+        });
+    },
+
+    /**
+     * Print the corresponding source label
+     */
+    _onClickPrintLabel: function(ev) {
+        // unfortunately, due to different reports needed for different models, we will handle
+        // pickings here and others models will have to be extended/printed separately until better
+        // technique to merge into continuous pdf to written
+        return this._printLabel(ev, 'stock.report_reception_report_label', 'stock.picking')
+    },
+
+    _printLabel: function(ev, reportFile, sourceType) {
+        let report_file = reportFile
+        let modelIds = [];
+        let productQtys = [];
+        let nodeToPrint = [];
+
+        if (! ev.currentTarget.className.includes('o_print_label_all')) {
+            nodeToPrint = [ev.currentTarget];
+        } else {
+            if (ev.target.name === "print_all_labels") {
+                nodeToPrint = this.iframe.contentDocument.querySelectorAll('.o_print_label:not(.o_print_label_all):not(:disabled)');
+            } else {
+                // Local print all
+                nodeToPrint = ev.target.closest('thead').nextElementSibling.querySelectorAll('.o_print_label:not(.o_print_label_all):not(:disabled)');
+            }
+        }
+
+        nodeToPrint.forEach(node => {
+            if (node.getAttribute('source-name') === sourceType) {
```

Comment:
IMO, the attribute `source-name` should be named as `source-model`, `model` or `res-model`, something who explicitly tells it's a model name, because `source-name` is confusing with the actual picking/MO's name.

Same for `sourceType`.